src/controllers/auth.controller.js

```diff
@@ -0,0 +1,201 @@
+import { validationResult } from 'express-validator';
+import bcrypt from 'bcrypt';
+import { v4 as uuidv4 } from 'uuid';
+
+import * as userService from '../services/user.service.js';
+import * as jwtService from '../services/jwt.service.js';
+import * as tokenService from '../services/token.service.js';
+import * as emailService from '../services/email.service.js';
+
+import ApiError from '../exeptions/api.error.js';
+import { hashPassword } from '../utils/helpers.js';
+
+const sendAuthentification = async (res, user, oldRefreshToken) => {
+  const secureUser = userService.secureUser(user);
+
+  const accessToken = jwtService.createToken(secureUser);
+  const refreshToken = jwtService.createRefreshToken(secureUser);
+
+  await tokenService.createToken(user.id, refreshToken);
+
+  if (oldRefreshToken) {
+    await tokenService.deleteTokenByToken(oldRefreshToken);
+  }
```

Comment:
All business logic should be in service, including interaction with other services.
In controllers, you do only things related to requests, such as calling some service, setting headers/cookies/status, sending response